src/httpserver.cpp

```diff
@@ -535,7 +527,7 @@ void StopHTTPServer()
     boundSockets.clear();
     {
         if (g_requests.CountActiveConnections() != 0) {
-            LogPrint(BCLog::HTTP, "Waiting for %d requests to stop HTTP server\n", g_requests.CountActiveRequests());
+            LogPrint(BCLog::HTTP, "Waiting for %d connections to stop HTTP server\n", g_requests.CountActiveConnections());
```

Comment:
to avoid grabbing the mutex twice in a row in the request tracker, could call `.CountActiveConnections` only once:
```
        if (auto connections = g_requests.CountActiveConnections(); connections != 0) {
            LogPrint(BCLog::HTTP, "Waiting for %d connections to stop HTTP server\n", connections);
        }
```

Reply:
Calling `CountActiveConnections()` just once will also avoid a weird message like "Waiting for 0 connections to stop HTTP server" which can happen if the function is called two times and the first call returns >0 and the second call returns 0 (the connection was closed in the meantime, after the first and before the second call).